src/main/java/frc/robot/subsystems/IntakeLimelight.java

```diff
@@ -364,8 +365,81 @@ public Optional<Double> getXOfSmallestY() {
     return Optional.of(corners[minYIdx - 1]);
   }
 
+  public Optional<Double> getObjectHeightPx(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    double minY = Double.MAX_VALUE;
+    double maxY = Double.MIN_VALUE;
+
+    for (int i = 1; i < corners.length; i += 2) {
+      minY = Math.min(minY, corners[i]);
+      maxY = Math.max(maxY, corners[i]);
+    }
+
+    return Optional.of(maxY - minY);
+  }
+
+  public class ConeDetection {
+    public double latencySec;
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Angle */
+    public double angleDeg;
+
+    public ConeDetection(double latencySec, double distanceMeters, double angleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.angleDeg = angleDeg;
+    }
+  }
+
+  /**
+   * Looks for a cone
+   *
+   * @return If empty, no cone detected. First value is cone yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<ConeDetection> getConePos() {
+    double[] corners = table.getEntry("tcornxy").getDoubleArray(new double[0]);
+    Optional<Double> maybeXPixels = getXOfSmallestY(corners);
+    Optional<Double> maybeYHeightPixels = getObjectHeightPx(corners);
+
+    if (!maybeXPixels.isPresent() || !maybeYHeightPixels.isPresent()) {
+      System.out.println("Didn't see cone");
+      return Optional.empty();
+    }
+
+    // Compute cone distance
+    final double vertFovDeg = 49.7;
+    double halfResYPixels = RESOLUTION_Y / 2.0;
+    double viewplaneHeightPixels = 2.0 * Math.tan(Units.degreesToRadians(vertFovDeg / 2));
+    double yPixels = maybeYHeightPixels.get();
+    double normYPixels = (1 / halfResYPixels) * yPixels;
+    double viewplaneYPixels = viewplaneHeightPixels / 2.0 * normYPixels;
+    final double CONE_HEIGHT_METERS = Units.inchesToMeters(13.0);
+    double coneDistanceMeters = CONE_HEIGHT_METERS / viewplaneYPixels;
```

Comment:
There was def somewhere we got this from... Are you able to find it and link it?

Reply:
wdym

Reply:
Added a link to the thingy above